src/data/polynomial/monic.lean

```diff
@@ -93,14 +93,28 @@ lemma monic_pow (hp : monic p) : ∀ (n : ℕ), monic (p ^ n)
 | 0     := monic_one
 | (n+1) := by { rw pow_succ, exact monic_mul hp (monic_pow n) }
 
-lemma monic_add_of_left {p q : R[X]} (hp : monic p) (hpq : degree q < degree p) :
+lemma monic_add_of_left (hp : monic p) (hpq : degree q < degree p) :
   monic (p + q) :=
 by rwa [monic, add_comm, leading_coeff_add_of_degree_lt hpq]
 
-lemma monic_add_of_right {p q : R[X]} (hq : monic q) (hpq : degree p < degree q) :
+lemma monic_add_of_right (hq : monic q) (hpq : degree p < degree q) :
   monic (p + q) :=
 by rwa [monic, leading_coeff_add_of_degree_lt hpq]
 
+lemma monic_of_mul_monic_left (hp : p.monic) (hpq : (p * q).monic) : q.monic :=
```

Comment:
Maybe `monic.of_mul_monic_left` or something similar is better, to allow dot notation?

Reply:
Yeah I do agree. In fact many lemmas in this file the same could be said for. I was thinking of doing that in a follow up PR.
Happy to change these ones when I get back home anyways.

Reply:
```suggestion
lemma monic.of_mul_monic_left (hp : p.monic) (hpq : (p * q).monic) : q.monic :=
```